runner/runner.go

```diff
@@ -0,0 +1,154 @@
+package runner
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"os/exec"
+	"syscall"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// NewRunner returns an initialized instance of *Runner.
+func NewRunner(role string) *Runner {
+	return &Runner{
+		role: role,
+	}
+}
+
+// Response provides a command response.
+type Response struct {
+	Body        []byte
+	Code        int
+	Name        string
+	Description string
+	FullCmd     []string
+	Timeout     time.Duration
+	Duration    time.Duration
+	Err         error
+}
+
+// BodyToString returns a string representation of combined stdout/stderr.
+func (r *Response) BodyToString() string {
+	return string(r.Body)
+}
+
+// Runner is a main instance of DC/OS runner runner.
+type Runner struct {
+	ClusterChecks map[string]*Check `json:"cluster_checks"`
+	NodeChecks    struct {
+		Checks    map[string]*Check `json:"checks"`
+		PreStart  []string          `json:"prestart"`
+		PostStart []string          `json:"poststart"`
+	} `json:"node_checks"`
+
+	role string
+}
+
+// LoadFromFile opens a config file and try to load the values to Runner struct.
+func (r *Runner) LoadFromFile(path string) error {
+	f, err := os.Open(path)
+	if err != nil {
+		return errors.Wrap(err, "unable to open config file")
+	}
+
+	if err := json.NewDecoder(f).Decode(r); err != nil {
+		return errors.Wrap(err, "unable to decode a config file")
+	}
+
+	return r.validate()
+}
+
+func (r *Runner) validate() error {
+	return nil
+}
+
+// Cluster executes cluster runner defined in config.
+func (r *Runner) Cluster(ctx context.Context, list bool, names ...string) ([]Response, error) {
+	return r.run(ctx, r.ClusterChecks, list, names...)
+}
+
+// PreStart executes the runner defined in config node_checks->prestart.
+func (r *Runner) PreStart(ctx context.Context, list bool, names ...string) ([]Response, error) {
+	checkNames := r.NodeChecks.PreStart
+	// if names is not empty override the prestart runner.
+	if len(names) > 0 {
+		checkNames = names
+	}
+	return r.run(ctx, r.NodeChecks.Checks, list, checkNames...)
+}
+
+// PostStart executes the runner defined in config node_checks->poststart.
+func (r *Runner) PostStart(ctx context.Context, list bool, names ...string) ([]Response, error) {
+	checkNames := r.NodeChecks.PostStart
+	// if names is not empty override the poststart runner.
+	if len(names) > 0 {
+		checkNames = names
+	}
+	return r.run(ctx, r.NodeChecks.Checks, list, checkNames...)
+}
+
+func (r *Runner) run(ctx context.Context, checkMap map[string]*Check, list bool, names ...string) ([]Response, error) {
+	// if not names passed, use all from checkMap
+	if len(names) == 0 {
+		for n := range checkMap {
+			names = append(names, n)
+		}
+	}
+
+	var responses []Response
+	for _, name := range names {
+		currentCheck, ok := checkMap[name]
+		if !ok {
```

Comment:
Safer to say `if !ok || currentCheck == nil` since the map is string -> pointer.